### PR TITLE
docs: add platform engineering projects 2022

### PR DESCRIPTION
**Pull request by milasuperstar**
Signed-off-by: Mila Wu <wumila1996@gmail.com>

### CODE REVIEW

**Comment by adela-bytebase** [comment]
@milasuperstar it's better to add links to Spotify, Apache Lucene and etc?

**Comment by milasuperstar** [comment]
> @milasuperstar maybe it's better to add links to Spotify, Apache Lucene and etc?
I added the link for lucene, methinks the readers can google the rest hahaha (don't want the post to be filled with links lmao

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+- API gateway - Kong
+- Universal Search - Elasticsearch
+- Code hosting - GitLab
+- Code Search - Sourcegraph
+- Monitoring/Alerting - Prometheus
+- Dashboard - Grafana
+- Task Scheduling Execution - Temporal
+- Keeping all of these resources in check - Terraform
+
+Now you have a fully private and deployable platform engineering system, but is there any other core components that have not been mentioned?
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:136]:
```suggestion
The above projects will constitute a pretty solid platform engineering backbone, while there are a couple of other core components worth mentioning.
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+- Security platform (login, VPN, Audit Log, intrusion detection)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:138]:
```suggestion
- Security platform (login, VPN, Audit Log, intrusion detection) - [Snyk](https://snyk.io/), [tailscale](https://tailscale.com/).
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+- Resource platform (expansion and contraction, scheduling, FinOps)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:139]:
```suggestion
- Resource platform (auto-scaling, scheduling, FinOps) [OpenCost](http://opencost.io/)
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+- Configuration platform (application configurations, feature flags)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:140]:
```suggestion
- Configuration platform (application configurations, feature flags) [Apollo](https://github.com/apolloconfig/apollo), [Flagsmith](https://flagsmith.com/).
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+- Database development platform (for database change & query management)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:141]:
```suggestion
- Database development platform (database change, query, admin, security management) - [Bytebase](https://bytebase.com)
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+Sidenote: if you are interested in database development platforms, we’d recommend [Bytebase](https://www.bytebase.com/). As for the other items, DM us [@StarHistoryHQ](https://twitter.com/StarHistoryHQ) and we can provide recommendations ;)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:143]:
```suggestion
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+In addition, the [Sourcegraph Podcast](https://about.sourcegraph.com/podcast), featuring co-founder/CTO Beyang and authors/engineers/DevRel folks of other startups, is an insightful podcast about projects' origin stories, these folks' worldviews, experiences, favourite tools and techs, and more.
+
+## Temporal
+
+![temporal](/blog/assets/yearly-pick-platform-engineering-2022/temporal.webp)
+
+[Temporal](https://github.com/temporalio/temporal) is another rising stat and serves as a core piece in platform engineering. It is a microservice task scheduling execution platform that executes application logic units, workflows in a resilient way, automatically handles intermittent failures, and retries failed operations. Although the first release only came in 2020.2, the founders built Cadence workflow (uber/cadence) back in Uber to solve Uber's workflow/orchestration problems, and Temporal is a commercialized fork version of Cadence.
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:108]:
```suggestion
[Temporal](https://github.com/temporalio/temporal) is another rising star and serves as a core piece in platform engineering. It is a microservice task scheduling execution platform that executes application logic units, workflows in a resilient way, automatically handles intermittent failures, and retries failed operations. Although the first release only came in 2020.2, the founders built Cadence workflow (uber/cadence) back in Uber to solve Uber's workflow/orchestration problems, and Temporal is a commercialized fork version of Cadence.
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+## Terraform
+
+![terraform](/blog/assets/yearly-pick-platform-engineering-2022/terraform.webp)
+
+[Terraform](https://github.com/hashicorp/terraform) is a tool for managing infrastructure (infrastructure-as-code) under HashiCorp. Terraform v0.1 was released back in 2014, but only became the industry leader 8 years later, revolutionizing the entire workflow, and was almost shut down in the early days because of the slow adoption.
+
+Terraform Provider started out as a way to smooth out the differences in managing the services of several large public cloud vendors, and has now become the de facto standard for managing all cloud/SaaS resources.
+
+## Platform engineering is the future
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:122]:
```suggestion
## Platform Engineering is the Future
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+Well, this is it for platform engineering projects, from big names to the starlets, let's review them again:
+
+- The underlying runtime platform - Kubernetes
+- The service catalog and developer portal - Backstage
+- API gateway - Kong
+- Universal Search - Elasticsearch
+- Code hosting and CI/CD - GitLab
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:130]:
```suggestion
- Code hosting, CI/CD, and DevSecOps - GitLab
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+- Dashboard - Grafana
+- Task Scheduling and Execution - Temporal
+- Managing all of the above resources (and the others) in code - Terraform
+
+The above projects will constitute a pretty solid platform engineering backbone, while there are a couple of other core components worth mentioning.
+
+- Security platform (login, VPN, Audit Log, intrusion detection) - [Snyk](https://snyk.io/), [tailscale](https://tailscale.com/).
+- Resource platform (auto-scaling, scheduling, FinOps) [OpenCost](http://opencost.io/)
+- Configuration platform (application configurations, feature flags) [Apollo](https://github.com/apolloconfig/apollo), [Flagsmith](https://flagsmith.com/).
+- Database development platform (database change, query, admin, security management) - [Bytebase](https://bytebase.com)
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:142]:
```suggestion
- Security and Access Control (network access, software supply chain, intrusion detection, Audit Log) - [Snyk](https://snyk.io/), [tailscale](https://tailscale.com/), [Vault](https://www.hashicorp.com/products/vault), [Boundary](https://www.boundaryproject.io/).
- Resource Allocation (auto-scaling, scheduling, FinOps) - [OpenCost](http://opencost.io/).
- Configuration (application configurations, feature flags) - [Apollo](https://github.com/apolloconfig/apollo), [Flagsmith](https://flagsmith.com/).
- Database Development (database change, query, admin, security) - [Bytebase](https://bytebase.com).
```

File: public/blog/star-history-yearly-pick-2022-platform-engineering.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+- The underlying runtime platform - Kubernetes
+- The service catalog and developer portal - Backstage
+- API gateway - Kong
+- Universal Search - Elasticsearch
+- Code hosting and CI/CD - GitLab
+- Code Search - Sourcegraph
+- Monitoring/Alerting - Prometheus
+- Dashboard - Grafana
+- Task Scheduling and Execution - Temporal
+- Managing all of the above resources (and the others) in code - Terraform
```
Comment (tianzhou) [inline public/blog/star-history-yearly-pick-2022-platform-engineering.md:135]:
```suggestion
- Manage all cloud resources and many of the above resources (and the others) in code - Terraform
```

File: public/blog/data.json [hunk 12]
```diff
@@ -1,4 +1,13 @@
 [
+  {
+    "title": "Star History Yearly Pick 2022 | Platform Engineering",
```
Comment (tianzhou) [inline public/blog/data.json:3]:
```suggestion
    "title": "Star History Open Source 2022 | Platform Engineering",
```

**Review by tianzhou — CHANGES_REQUESTED** [review]

**Review by tianzhou — APPROVED** [review]
LGTM

**Pull request merged**